core-bundle/src/Command/UserListCommand.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Command;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\Model\Collection;
+use Contao\UserModel;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Lists Contao back end users.
+ *
+ * @internal
+ */
+class UserListCommand extends Command
+{
+    protected static $defaultName = 'contao:user:list';
+
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    public function __construct(ContaoFramework $framework)
+    {
+        $this->framework = $framework;
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addOption('column', null, InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY, 'The columns display in the table')
+            ->addOption('admins', null, InputOption::VALUE_NONE, 'Return only admins')
+            ->addOption('format', null, InputOption::VALUE_REQUIRED, 'The output format', 'text')
+            ->setDescription('Lists Contao back end users.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        if ($input->getOption('admins')) {
+            $io->note('Only showing admin accounts');
+        }
+
+        $users = $this->getUsers($input->getOption('admins'));
+
+        if (null === $users) {
+            $io->error('No accounts found.');
+
+            return 1;
+        }
+
+        $columns = $input->getOption('column');
+
+        switch ($input->getOption('format')) {
+            case 'text':
+                $rows = $this->formatTableRows($users, $columns);
+
+                $io->table($columns, $rows);
+                break;
+
+            case 'json':
+                $data = $this->formatJson($users, $columns);
+
+                $io->write(json_encode($data));
+                break;
+
+            default:
+                throw new \LogicException('Invalid format: '.$input->getOption('format'));
+        }
+
+        return 0;
+    }
+
+    private function getUsers(bool $onlyAdmins = false): ?Collection
+    {
+        $this->framework->initialize();
+
+        /** @var UserModel $userModel */
+        $userModel = $this->framework->getAdapter(UserModel::class);
+
+        if ($onlyAdmins) {
+            return $userModel->findBy('admin', '1');
+        }
+
+        return $userModel->findAll();
+    }
+
+    private function formatTableRows(Collection $users, array &$columns): array
+    {
+        if ([] === $columns) {
```

Comment:
should we add some sort of protection to dump the password from the database?

Reply:
The password hashes are useless and less valuable than the email IMHO.

Reply:
I'm not sure about that. @ausi /cc

Reply:
I agree with @richardhj. The email address of a user is potentially a more sensitive value than the password hash. Also, if you have access to this command, you could always just connect to the database directly and look the password hash up, right?

And looking at the password hashes might be useful to see if all users upgraded to a new hashing algorithm for example.